internal/api/health.go

```diff
@@ -3,17 +3,26 @@ package api
 import (
 	"context"
 	"net/http"
+	"sync"
+	"time"
 
 	"github.com/gin-gonic/gin"
 )
 
 type Health struct {
 	TerminatingStatusCode int `description:"Terminating status code" json:"terminatingStatusCode,omitempty" yaml:"terminatingStatusCode,omitempty" export:"true"`
 	terminating           bool
+	dockerPingFunc        func(context.Context) error
```

Comment:
I would change to a non-specific provider func.

Everything related to docker specifically should be drop